pkg/operator/operatorclient/interfaces.go

```diff
@@ -83,3 +88,158 @@ func (c *CLIManagerClient) GetObjectMeta() (meta *metav1.ObjectMeta, err error)
 	}
 	return &instance.ObjectMeta, nil
 }
+
+func (c *CLIManagerClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, desiredConfiguration *applyconfiguration.OperatorSpecApplyConfiguration) error {
+	if desiredConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	desired := climanagerapplyconfiguration.CliManager(OperatorConfigName, OperatorNamespace)
+	desiredSpec := climanagerapplyconfiguration.CliManagerSpec()
+	if desiredConfiguration.LogLevel != nil {
+		desiredSpec.WithLogLevel(*desiredConfiguration.LogLevel)
+	}
+	if desiredConfiguration.ManagementState != nil {
+		desiredSpec.WithManagementState(*desiredConfiguration.ManagementState)
+	}
+	if desiredConfiguration.UnsupportedConfigOverrides != nil {
+		desiredSpec.WithUnsupportedConfigOverrides(*desiredConfiguration.UnsupportedConfigOverrides)
+	}
+	if desiredConfiguration.OperatorLogLevel != nil {
+		desiredSpec.WithOperatorLogLevel(*desiredConfiguration.OperatorLogLevel)
+	}
+	if desiredConfiguration.ObservedConfig != nil {
+		desiredSpec.WithObservedConfig(*desiredConfiguration.ObservedConfig)
+	}
+	desired.WithSpec(desiredSpec)
```

Comment:
https://github.com/openshift/library-go/blob/756adf2188fcdb8f4af531514a76bc7bded706b7/pkg/operator/genericoperatorclient/dynamic_operator_client.go#L229-L271 has a use of `extractApplySpec`. Something David introduced recently through https://github.com/openshift/library-go/pull/1787. Worth checking whether it makes sense to actually require to pass extractApplySpec instead of going through the fields one by one. It's a common code that will be used by all operators embedding OperatorSpec.

Reply:
I tried to use [this function](https://github.com/openshift/library-go/blob/756adf2188fcdb8f4af531514a76bc7bded706b7/pkg/operator/genericoperatorclient/dynamic_operator_client.go#L30) but it accepts `unstructured.Unstructured` (that we can pass) but returns `StaticPodOperatorSpecApplyConfiguration` (we need to have `CliManagerApplyConfiguration`) that is not compatible with what we need.

Reply:
Instead of manual one by one mapping (and error prone), json serialize/deserialize is used that is inspired from https://github.com/openshift/library-go/blob/cf085edda2daa591467bf0075e7c7e8d2ab1b8bf/pkg/operator/v1helpers/canonicalize.go#L18-L33.

@ingvagabund the latest version of this PR looks like more production ready. Could you PTAL?.